src/use-hover/use-hover.ts

```diff
@@ -0,0 +1,40 @@
+import { method } from '../support/index'
+import { HoverController } from './hover-controller'
+
+export interface HoverOptions {
+  element?: Element
+}
+
+export const useHover = (controller: HoverController, options: HoverOptions = {}) => {
+  const targetElement: Element = options?.element || controller.element
+
+  const onEnter = () => {
+    controller.mouseEnter && method(controller, 'mouseEnter').call(controller)
+  }
+  const onLeave = () => {
+    controller.mouseLeave && method(controller, 'mouseLeave').call(controller)
+  }
+
+  const controllerDisconnect = controller.disconnect.bind(controller)
+
+  const observe = () => {
+    targetElement.addEventListener('mouseenter', onEnter)
+    targetElement.addEventListener('mouseleave', onLeave)
+  }
+
+  const unobserve = () => {
+    targetElement.removeEventListener('mouseenter', onEnter)
+    targetElement.removeEventListener('mouseleave', onLeave)
+  }
+
+  Object.assign(controller, {
+    disconnect() {
+      unobserve()
+      controllerDisconnect()
+    }
+  })
+
+  observe()
+
+  return [observe, unobserve] as const
+}
```

Comment:
Thanks for trying to match the current code style. 

I have recently introduced a debug mode (#85) and as part of the debug mode start to refactor a bit the code to rely more on `Classes` rather than pure functions. There is a base class `StimulusUse` that provides a few common features (currently mostly the log function). 
I am in the process of converting the different mixins to this pattern, but it is still a WIP. Currently, `useDispatch` & `useVisibility` are based on this. It would be great if `useHover` could rely on this pattern out of the box.

I haven't tested it but here is a quick conversion of the code to this format

```js
import { method } from '../support/index'
import { StimulusUse, StimulusUseOptions } from '../stimulus_use'
import { HoverController } from './hover-controller'

export interface HoverOptions extends StimulusUseOptions {
  element?: Element
}

export class UseHover extends StimulusUse {
  controller: HoverController
  targetElement: Element

  constructor(controller: HoverController, options: HoverOptions = {}) {
    super(controller, options)
    this.targetElement = options?.element || controller.element
    this.controller = controller
    this.enhanceController()
    this.observe()
  }

  observe = () => {
    this.targetElement.addEventListener('mouseenter', this.onEnter)
    this.targetElement.addEventListener('mouseleave', this.onLeave)
  }

  unobserve = () => {
    this.targetElement.removeEventListener('mouseenter', this.onEnter)
    this.targetElement.removeEventListener('mouseleave', this.onLeave)
  }

  // private

  private onEnter = () => {
    this.controller.mouseEnter && method(this.controller, 'mouseEnter').call(this.controller)
    this.log("mouseEnter", { hover: true })
  }

  private onLeave = () => {
    this.controller.mouseLeave && method(this.controller, 'mouseLeave').call(this.controller)
    this.log("mouseLeave", { hover: false })
  }

  private enhanceController() {
    const controllerDisconnect = this.controller.disconnect.bind(this.controller)

    const disconnect = () => {
      this.unobserve()
      controllerDisconnect()
    }

    Object.assign(this.controller, { disconnect })
  }
}

export const useHover = (controller: HoverController, options: HoverOptions = {}) => {
  const observer = new UseHover(controller, options)
  return [observer.observe, observer.unobserve] as const
}
```

Reply:
Ooh nice, seems to work in testing, pushed those changes :) 👍